core/src/syscall/host.rs

```diff
@@ -207,3 +207,37 @@ where
     None
   }
 }
+
+pub struct SyscallHostSpawn;
+
+impl SyscallHostSpawn {
+  pub const fn new() -> Self {
+    Self
+  }
+}
+
+impl<T> Syscall<T> for SyscallHostSpawn
+where
+  T: HostInterface,
+{
+  fn execute(&self, ctx: &mut SyscallContext<T>, arg1: u32, arg2: u32) -> Option<u32> {
+    // let athena_ctx = ctx
+    //   .rt
+    //   .context
+    //   .as_ref()
+    //   .expect("Missing Athena runtime context");
+
+    let len = arg2;
+    let blob: Vec<u8> = ctx
+      .slice(arg1, len as usize / 4)
```

Comment:
What if `len` isn't a multiple of 4?

Reply:
As it stands, we require the template to pad the blob. I think this is okay for now.

Reply:
Assuming you're okay with this, let me know if not. This is one of many things we can revisit in the future to make the devex better.